pkg/daemon/update.go

```diff
@@ -680,13 +687,36 @@ func (dn *Daemon) writeFiles(files []igntypes.File) error {
 				return fmt.Errorf("failed to retrieve file ownership for file %q: %v", file.Path, err)
 			}
 		}
+		if err := createOrigFile(file.Path); err != nil {
+			return err
+		}
 		if err := writeFileAtomically(file.Path, contents.Data, defaultDirectoryPermissions, mode, uid, gid); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func origFileName(fpath string) string {
+	return fpath + ".mcdorig"
+}
+
+func createOrigFile(fpath string) error {
+	if _, err := os.Stat(fpath); err != nil {
+		// the file isn't there, no need to back it up
+		// we could check ENOENT only maybe?
+		return nil
+	}
+	if _, err := os.Stat(origFileName(fpath)); err == nil {
+		// the orig file is already there and we avoid creating a new one to preserve the real default
+		return nil
+	}
+	if out, err := exec.Command("cp", "-a", "--reflink=auto", fpath, origFileName(fpath)).CombinedOutput(); err != nil {
```

Comment:
Of course if this is interrupted in the middle we'll end up with an incomplete backup but...let's just get this in!

Reply:
The [libglnx code](https://gitlab.gnome.org/GNOME/libglnx/blob/8f34033b256da5a8e2cc66faf8e81cfdfb9cdf37/glnx-fdio.c#L930) I wrote for this properly clones to an `O_TMPFILE` which also makes it superior to what the Rust libstd is doing.  It's a bit surprising actually `cp` doesn't have a `--tmpfile` flag or something, would probably be easy to implement.

If only calling C from golang wasn't unspeakably awful...

Anyways we can circle back to polishing this bit later.